web/wp-content/mu-plugins/pantheon/pantheon-updates.php

```diff
@@ -0,0 +1,151 @@
+}
+
+// Check if Pantheon upstream updates are available.
+function _pantheon_wordpress_update_available() {
+
+	if ( ! function_exists( 'pantheon_curl' ) ) {
+		return false;
+	}
+
+	/**
+	 * If the site is using the default WordPress upstream and
+	 * WordPress is up to date, do not show the update notice
+	 */
+	if( ! _pantheon_fetch_custom_upstream_info() && _pantheon_is_wordpress_core_latest() ) {
+		return false;
+	}
+
+	$upstream_updates_api_url = 'https://api.live.getpantheon.com/sites/self/code-upstream-updates';
+	if ( 'dev' != $_ENV['PANTHEON_ENVIRONMENT'] ) {
+		$upstream_updates_api_url .= '?base_branch=refs%2Fheads%2F'.$_ENV['PANTHEON_ENVIRONMENT'];
+	}
+
+	$data = _pantheon_curl_cached( $upstream_updates_api_url );
+	if ( empty( $data['update_log'] ) ) {
+		return false;
+	}
+	return true;
+}
+
+function _pantheon_curl_cached( $api_url ) {
+	$cache_key   = 'pantheon_curl_' . md5( $api_url );
+	$cache_value = get_transient( $cache_key );
+	if ( false !== $cache_value ) {
+		return $cache_value;
+	}
+	$api_response = pantheon_curl( $api_url, null, 8443 );
+	$data = $api_response ? json_decode( $api_response['body'], true ) : [];
+	set_transient( $cache_key, $data, 2 * MINUTE_IN_SECONDS );
+	return $data;
+}
+
+// Replace WordPress core update nag EVERYWHERE with our own notice (use git upstream)
+function _pantheon_upstream_update_notice() {
+	$update_type = 'new WordPress version';
+	$update_help = 'If you need help, open a support chat on Pantheon.';
+	$upstream_path = _pantheon_fetch_custom_upstream_info();
+	if ( ! empty( $upstream_path ) ) {
+		$update_type = 'Pantheon Custom Upstream update from "'.$upstream_path.'"';
+		$update_help = 'If you need help, contact an administrator for your Pantheon organization.';
+	}
+
+    ?>
+    <div class="update-nag">
+		<p style="font-size: 14px; font-weight: bold; margin: 0 0 0.5em 0;">
+			A <?php echo $update_type; ?> is available! Please update from <a href="https://dashboard.pantheon.io/sites/<?php echo $_ENV['PANTHEON_SITE']; ?>">your Pantheon dashboard</a>.
+		</p>
+		For details on applying updates, see the <a href="https://pantheon.io/docs/upstream-updates/" target="_blank">Applying Upstream Updates</a> documentation. <br />
+		<?php echo $update_help; ?>
+	</div>
+    <?php
+}
+
+// Register Pantheon specific WordPress update admin notice
+add_action( 'admin_init', '_pantheon_register_upstream_update_notice' );
+function _pantheon_register_upstream_update_notice(){
+	// but only if we are on Pantheon
+	// and there is a WordPress update available
+	if( isset( $_ENV['PANTHEON_ENVIRONMENT'] ) && _pantheon_wordpress_update_available() ){
+		add_action( 'admin_notices', '_pantheon_upstream_update_notice' );
+	}
+}
+
+// Return zero updates and current time as last checked time
+function _pantheon_disable_wp_updates() {
+	include ABSPATH . WPINC . '/version.php';
+	return (object) array(
+		'updates' => array(),
+		'version_checked' => $wp_version,
+		'last_checked' => time(),
```

Comment:
:warning: **Warning**: Array double arrow not aligned correctly; expected 4 space(s) between &quot;'last_checked'&quot; and double arrow, but found 1 (*WordPress.Arrays.MultipleStatementAlignment.DoubleArrowNotAligned*).